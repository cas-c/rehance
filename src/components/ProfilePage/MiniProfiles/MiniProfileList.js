import React from 'react';
import PropTypes from 'prop-types';
import MiniProfile from './MiniProfile';

const MiniProfiles = ({ users }) => users.slice(0, 5).map((u, i) => <MiniProfile user={u} key={`mini-user-${i}`} />);

MiniProfiles.propTypes = {
    users: PropTypes.arrayOf(PropTypes.object)
};

MiniProfiles.defaultProps = {
    users: []
}

export default MiniProfiles;
