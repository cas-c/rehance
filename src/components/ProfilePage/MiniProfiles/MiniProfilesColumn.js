import React from 'react';
import PropTypes from 'prop-types';
import MiniProfileList from './MiniProfileList';

const MiniProfilesColumn = ({ label, data, emptyString }) => (
    <div className='column'>
        <div className='is-size-3 profile-page-header'>
            { label }
        </div>
        {
            data.length > 0 ?
                <MiniProfileList users={data} /> :
                <div className='has-text-centered'>{ emptyString }</div>
        }
        { data.length > 5 && <div className='has-text-right'><a>View all...</a></div> }
    </div>
);

MiniProfilesColumn.propTypes = {
    label: PropTypes.string,
    emptyString: PropTypes.string,
    data: PropTypes.arrayOf(PropTypes.object)
};

MiniProfilesColumn.defaultProps = {
    label: '',
    emptyString: 'Nothing to display.',
    data: []
};

export default MiniProfilesColumn;