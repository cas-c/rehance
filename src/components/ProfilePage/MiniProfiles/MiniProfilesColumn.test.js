import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import MiniProfilesColumn from './MiniProfilesColumn';
import MiniProfileList from './MiniProfileList';

const mockUsers = [
    {
        images: {
            115: 'testImageUrl'
        },
        display_name: 'Test User',
        username: 'miniprofileuser'
    },
    {
        images: {
            115: 'testImageUrl2'
        },
        display_name: 'Test User 2',
        username: 'miniprofileuser2'
    }
];

const mockUsersMoreThanFive = [
    {
        images: {
            115: 'testImageUrl'
        },
        display_name: 'Test User',
        username: 'miniprofileuser'
    },
    {
        images: {
            115: 'testImageUrl2'
        },
        display_name: 'Test User 2',
        username: 'miniprofileuser2'
    },
    {
        images: {
            115: 'testImageUrl3'
        },
        display_name: 'Test User 3',
        username: 'miniprofileuser3'
    },
    {
        images: {
            115: 'testImageUrl4'
        },
        display_name: 'Test User 4',
        username: 'miniprofileuser4'
    },
    {
        images: {
            115: 'testImageUrl5'
        },
        display_name: 'Test User 5',
        username: 'miniprofileuser5'
    },
    {
        images: {
            115: 'testImageUrl6'
        },
        display_name: 'Test User 6',
        username: 'miniprofileuser6'
    },
    {
        images: {
            115: 'testImageUrl7'
        },
        display_name: 'Test User 7',
        username: 'miniprofileuser7'
    }
];

describe('MiniProfileList', () => {
    it('snapshot: should render correctly with input (less than / equal to five)', () => {
        const output = shallow(<MiniProfilesColumn label='Testing' emptyString='No input.' users={mockUsers}/>);
        expect(shallowToJson(output)).toMatchSnapshot();
    });
    it('snapshot: should render correctly with input (more than five)', () => {
        const output = shallow(<MiniProfilesColumn label='Testing' emptyString='No input.' users={mockUsersMoreThanFive}/>);
        expect(shallowToJson(output)).toMatchSnapshot();
    });
    it('snapshot: should render correctly without input', () => {
        const output = shallow(<MiniProfilesColumn />);
        expect(shallowToJson(output)).toMatchSnapshot();
    });
    it('snapshot: should render correctly without input and with custom empty string', () => {
        const output = shallow(<MiniProfilesColumn label='Testing' emptyString='Custom empty string.' />);
        expect(shallowToJson(output)).toMatchSnapshot();
    });
    it('unit: should display the emptyString if no/empty data is entered', () => {
        const mounted = mount(<MiniProfilesColumn emptyString='Custom string' />);
        expect(mounted.containsMatchingElement(<div>Custom string</div>)).toBe(true);
        mounted.unmount();
        const mountedWithEmpty = mount(<MiniProfilesColumn emptyString='Custom string' data={[]} />);
        expect(mountedWithEmpty.containsMatchingElement(<div>Custom string</div>)).toBe(true);
        mountedWithEmpty.unmount();
    });
    it('unit: displays a "View all..." link only if there are more than five users', () => {
        const mounted = mount(<MiniProfilesColumn data={mockUsersMoreThanFive} />);
        expect(mounted.containsMatchingElement(<a>View all...</a>)).toBe(true);
        mounted.unmount();
        const mountedLessThan = mount(<MiniProfilesColumn data={mockUsers} />);
        expect(mountedLessThan.containsMatchingElement(<a>View all...</a>)).toBe(false);
        mountedLessThan.unmount();
    });
})