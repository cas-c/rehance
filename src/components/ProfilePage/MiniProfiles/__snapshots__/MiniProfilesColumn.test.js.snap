// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MiniProfileList snapshot: should render correctly with input (less than / equal to five) 1`] = `
<div
  className="column"
>
  <div
    className="is-size-3 profile-page-header"
  >
    Testing
  </div>
  <div
    className="has-text-centered"
  >
    No input.
  </div>
</div>
`;

exports[`MiniProfileList snapshot: should render correctly with input (more than five) 1`] = `
<div
  className="column"
>
  <div
    className="is-size-3 profile-page-header"
  >
    Testing
  </div>
  <div
    className="has-text-centered"
  >
    No input.
  </div>
</div>
`;

exports[`MiniProfileList snapshot: should render correctly without input 1`] = `
<div
  className="column"
>
  <div
    className="is-size-3 profile-page-header"
  />
  <div
    className="has-text-centered"
  >
    Nothing to display.
  </div>
</div>
`;

exports[`MiniProfileList snapshot: should render correctly without input and with custom empty string 1`] = `
<div
  className="column"
>
  <div
    className="is-size-3 profile-page-header"
  >
    Testing
  </div>
  <div
    className="has-text-centered"
  >
    Custom empty string.
  </div>
</div>
`;
